src/devTools.js

```diff
@@ -50,13 +50,17 @@ function connectToServer(options) {
   watch();
 }
 
-export function start(options) {
+export async function start(options, hostnamePromise) {
   if (options) {
-    if (options.port && !options.hostname) {
-      options.hostname = 'localhost';
+    try {
+      if (options.port && !options.hostname) {
+        options.hostname = await hostnamePromise;
+      }
+      connectToServer(options);
```

Comment:
I think the `connectToServer` call should be moved outside of the `try-catch`, since an error there does not count as an "Error obtaining socket hostname", and the changes would be more minimal if it we leave it outside the `try-catch`